File: Interpreted/man-or-boy/mob.py
#!/usr/bin/env python

import sys
sys.setrecursionlimit(40000)

def a(k, x1, x2, x3, x4, x5):
    def b():
        b.k -= 1
        return a(b.k, b, x1, x2, x3, x4)
    b.k = k
    return x4() + x5() if b.k <= 0 else b()

x = lambda i: lambda: i
print(a(int(sys.argv[1]), x(1), x(-1), x(-1), x(1), x(0)))

File: Interpreted/man-or-boy/mob.rb
def a(k, x1, x2, x3, x4, x5)
  b = lambda { k -= 1; a(k, b, x1, x2, x3, x4) }
  k <= 0 ? x4[] + x5[] : b[]
end

puts a(ARGV[0].to_i, lambda {1}, lambda {-1}, lambda {-1}, lambda {1}, lambda {0})
